docarray/doc_index/backends/elastic_doc_index.py

```diff
@@ -0,0 +1,420 @@
+import uuid
+import warnings
+from dataclasses import dataclass, field
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Dict,
+    Generator,
+    Generic,
+    Iterable,
+    List,
+    Mapping,
+    Optional,
+    Sequence,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+    cast,
+)
+
+import numpy as np
+from elasticsearch import Elasticsearch
+from elasticsearch.helpers import parallel_bulk
+
+import docarray.typing
+from docarray import BaseDocument, DocumentArray
+from docarray.doc_index.abstract_doc_index import (
+    BaseDocumentIndex,
+    FindResultBatched,
+    _ColumnInfo,
+)
+from docarray.utils.find import FindResult
+from docarray.utils.misc import torch_imported
+
+# mypy: ignore-errors
+
+if TYPE_CHECKING:
+
+    def composable(fn):  # type: ignore
+        return fn
+
+else:
+    # static type checkers do not like callable objects as decorators
+    from docarray.doc_index.abstract_doc_index import composable
+
+TSchema = TypeVar('TSchema', bound=BaseDocument)
+T = TypeVar('T', bound='ElasticDocumentIndex')
+
+MAX_ES_RETURNED_DOCS = 10000
```

Comment:
This is the default value. The user can choose to increase the [index.max_result_window](https://www.elastic.co/guide/en/elasticsearch/reference/8.6/index-modules.html#dynamic-index-settings) to a different value.